source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -37,13 +37,17 @@ Http::FilterHeadersStatus CacheFilter::decodeHeaders(Http::RequestHeaderMap& hea
         *decoder_callbacks_, headers);
     return Http::FilterHeadersStatus::Continue;
   }
-  if (!CacheFilterUtils::isCacheableRequest(headers)) {
+  if (!CacheabilityUtils::isCacheableRequest(headers)) {
     ENVOY_STREAM_LOG(debug, "CacheFilter::decodeHeaders ignoring uncacheable request: {}",
                      *decoder_callbacks_, headers);
     return Http::FilterHeadersStatus::Continue;
   }
   ASSERT(decoder_callbacks_);
-  lookup_ = cache_.makeLookupContext(LookupRequest(headers, time_source_.systemTime()));
+
+  LookupRequest lookup_request(headers, time_source_.systemTime());
+  request_allows_inserts_ = !lookup_request.requestCacheControl().no_store_;
```

Comment:
suggest leaving link to http caching spec describing the semantics; I'm not sure if absence of "no-store" on request is all that we need to know we haven't disallowed saving cache from the request. It probably is but a spec ref would be awesome.

Reply:
I will add the relevant RFC link at the bool definition.